guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/upload/UploadForm.java

```diff
@@ -107,13 +110,18 @@ public void onSubmit( final Form.SubmitEvent event ) {
                     BusyPopup.close();
                     Window.alert( M2RepoEditorConstants.INSTANCE.SelectFileUpload() );
                     event.cancel();
+                } else if ( !( isJar( fileName ) || isKJar( fileName ) || isPom( fileName ) ) ) {
```

Comment:
what about extract a method like, and just use if ( !isValid( fileName ) )
 
    private boolean isValid( final String fileName ) {
        return isJar( fileName ) || isKJar( fileName ) || isPom( fileName );
    }